tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -108,7 +109,61 @@ def h_api(self, pyramid_config):
         return svc
 
 
-class TestCanvasFileBasicLTILaunch:
+class ConfiguredLaunch:
+    method_to_test = None
```

Comment:
I don't think you can do that here; at least not like that. The missing piece is the fact that each of the child classes override `pyramid_request` to setup their own parameters on it.

I don't see a way of dealing with that in a concrete test class without copy and pasting the request fixtures into that class. As the point of this was to reduce copy paste that would be counter productive. Most of the ways of making sure the correct methods are paired with the correct request objects are very ugly as well.

Other solutions to this I tried involved class method assert functions you call which you pass a lambda which is just a function which calls the `method_to_test`. The same thing with a context manager. Both of these suffer from the same problem: It doesn't really save you any code, you are repeating the same method functions over and over. There will be 12 stub calls which don't do anything informative.

With regards to other objections to the inheritance solution: 
 * `_call_method()` is probably needlessly magic. I think I'll replace it with an unimplemented method the children must implement. I was trying to make the child classes as short and declarative as possible. This was the price.
 * `method_to_test` isn't used in the child classes - True, and it probably should be as they all repeatedly call the function. I think this will be more obvious with a call method into the children as above.
 * I don't think a child class is going to accidentally re-implement a function called `test_it_does_not_call_list_result_upsert_if_instructor`, but perhaps. It's a possibility, but it feels like a bit of a reach. On the other hand, if a child wants to re-implement it, they can.

I'm totally happy to do this in a different pytest fixtured way if it's possible, but I think that solution would need to:

 * Not copy many fixtures between different tests
 * Not create many floating seemingly global fixtures which are actually local but homeless now
 * Not introduce lots of placeholder functions that just call another function

Reply:
I think this might also be hinting at what I was talking about. If these three views had a base-class, or were just separated into a distinct class, that class would have concrete tests, and this would be the natural home for these tests.

I think three views that all have the same series of 5 calls or so probably indicates they should be separated from the other ones.

Reply:
I'm still not sure whether inheritance rather than composition is the right solution. But I do agree that the fact that we're having to use test class inheritance or parametrized fixtures or other test code cleverness in order to test this effectively ... Indicates that there's a code design problem.

Anyway, sounds like the parametrized fixture is a PITA, so let's just stick with our base test class approach